bigquery_etl/cli/stage.py

```diff
@@ -260,13 +261,51 @@ def _udf_dependencies(artifact_files):
 def _view_dependencies(artifact_files, sql_dir):
     """Determine view dependencies."""
     view_dependencies = set()
-    view_dependency_files = [file for file in artifact_files if file.name == VIEW_FILE]
+    view_dependency_files = [
+        file for file in artifact_files if file.name in {VIEW_FILE, QUERY_FILE}
+    ]
     for dep_file in view_dependency_files:
         # all references views and tables need to be deployed in the same stage project
         if dep_file not in artifact_files:
             view_dependencies.add(dep_file)
 
-        if dep_file.name == VIEW_FILE:
+        if dep_file.name == QUERY_FILE:
+            # check query files for view dependencies
+            # views need to be deployed to stage as they might reference other changed artifacts upstream
+            # without getting changed themselves
+            dependencies = extract_table_references(
+                render(dep_file.name, dep_file.parent)
+            )
+            for dependency in dependencies:
+                dependency_components = dependency.split(".")
+
+                if len(dependency_components) == 3:
```

Comment:
Could this be a good place to use `match`?
https://docs.python.org/3/tutorial/controlflow.html#match-statements